frontend/src/context/AuthContext.jsx

```
@@ -14,40 +49,285 @@
 
   // Check authentication status on app load
   useEffect(() => {
     checkAuthStatus();
-  }, []);
 
-  const checkAuthStatus = async () => {
+    // Cleanup on unmount
+    return () => {
+      stopTokenMonitoring();
+    };
+  }, [checkAuthStatus, stopTokenMonitoring]);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)